Project.toml

```diff
@@ -9,6 +9,7 @@ ColorTypes = "3da002f7-5984-5a60-b8a6-cbb66c0b333f"
 Colors = "5ae59095-9a9b-59fe-a467-6f913c188581"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+LRUCache = "8ac3fa9e-de4c-5943-b1dc-09c6b5f20637"
```

Comment:
This might also be a weak dependency?

Reply:
That is now really a case where I think it is a tradeoff between ease of use (“Caches just work”) and that you have to load this, I can think about it. If so, I will probably even rework all dependencies into the way Chris did in DiffEq, to be weak for 1.9 and Require earlier.

Reply:
The only user-facing difference would be whether they have to type `using LRUCache` or not before using caches, and for people who don't need caches the package would be lighter.

BTW, `Manifolds`, `Plots` and `LineSearches` could also be turned into weak dependencies.

Reply:
Yes I would do that on the same run. And we should check Manifold for that as well.

Ok, maybe weak is a good idea, will do it along the way when most of the modelling is done.

Reply:
I'm going to try doing some weak dependencies for Manifolds.jl soonish.

Reply:
This is the guide I will follow https://discourse.julialang.org/t/package-extensions-for-julia-1-9/93397/5 (but I think you did that as well somewhere already?

Reply:
Yes, I've seen that guide, it is good. I haven't used it fully yet though because I've just switched to 1.9 a couple of days ago but I've followed the general file structure in the LineSearches PR here.